### ISSUE TITLE
[Text][Image][Teaser] In-place editing config path is not accessible for non-administrators

### ISSUE THREAD

**Issue by vladbailescu** [adobe/aem-core-wcm-components#657]
## Bug Report

**Non-administrator users do not have access to the current in-place editing configuration**
Because of a global ACL that forbids not-admin users to access OSGi configs (ACL is for `/apps/*/config/*`), normal users cannot read the in-place editing configuration for:

* Text:
  * https://github.com/adobe/aem-core-wcm-components/blob/master/content/src/content/jcr_root/apps/core/wcm/components/text/v1/text/_cq_editConfig.xml#L8
  * https://github.com/adobe/aem-core-wcm-components/blob/master/content/src/content/jcr_root/apps/core/wcm/components/text/v2/text/_cq_editConfig.xml#L8
* Title
  * https://github.com/adobe/aem-core-wcm-components/blob/master/content/src/content/jcr_root/apps/core/wcm/components/title/v1/title/_cq_editConfig.xml#L8
  * https://github.com/adobe/aem-core-wcm-components/blob/master/content/src/content/jcr_root/apps/core/wcm/components/title/v2/title/_cq_editConfig.xml#L8
* Image
  * https://github.com/adobe/aem-core-wcm-components/blob/master/content/src/content/jcr_root/apps/core/wcm/components/image/v1/image/_cq_editConfig.xml#L21
  * https://github.com/adobe/aem-core-wcm-components/blob/master/content/src/content/jcr_root/apps/core/wcm/components/image/v2/image/_cq_editConfig.xml#L21
* Teaser
  * https://github.com/adobe/aem-core-wcm-components/blob/master/content/src/content/jcr_root/apps/core/wcm/components/teaser/v1/teaser/_cq_editConfig.xml#L21

**Expected behavior/code**
In-place editing configuration is available for all users

**Possible Solution**
We can move the configuration node and set a `configPath` to point to its location, see https://helpx.adobe.com/experience-manager/6-3/sites/administering/using/rich-text-editor.html#Understandtheconfigurationpathsandlocations

**Comment by ahmed-musallam** [comment]
@vladbailescu I am currently seeing this issue in AMS Cloud Manager. Is there anyway to get around it with version 2.4.0? or 2.5.0 ?

**Comment by ahmed-musallam** [comment]
Posting here for anyone else that runs into this issue with AMS:

> THIS IS A TEMPORARY FIX UNTILL CORE COMPONENTS 2.6.0 IS RELEASED.

I have core components package as a subpackage in my application package. (tried both versions 2.4.0 and 2.5.0).
On AMS build, this returns the following errors in a CSV report:


File Location | Line Number | Issue | Type | Severity | Effort | Rule | Tags
-- | -- | -- | -- | -- | -- | -- | --
adobe/cq60:core.wcm.components.content:2.4.0 | 0 | Node   /apps/core/wcm/components/teaser/v1/teaser/cq:editConfig/cq:inplaceEditing/config   is an OSGi config or install path which contains non-OSGi-related children   that will not be visible to non-administrative users. | Bug | Major |   | ConfigAndInstallShouldOnlyContainOsgiNodes | aem
adobe/cq60:core.wcm.components.content:2.4.0 | 0 | Node   /apps/core/wcm/components/image/v1/image/cq:editConfig/cq:inplaceEditing/config   is an OSGi config or install path which contains non-OSGi-related children   that will not be visible to non-administrative users. | Bug | Major |   | ConfigAndInstallShouldOnlyContainOsgiNodes | aem
adobe/cq60:core.wcm.components.content:2.4.0 | 0 | Node   /apps/core/wcm/components/image/v2/image/cq:editConfig/cq:inplaceEditing/config   is an OSGi config or install path which contains non-OSGi-related children   that will not be visible to non-administrative users. | Bug | Major |   | ConfigAndInstallShouldOnlyContainOsgiNodes | aem
adobe/cq60:core.wcm.components.content:2.4.0 | 0 | Node   /apps/core/wcm/components/text/v2/text/cq:editConfig/cq:inplaceEditing/config   is an OSGi config or install path which contains non-OSGi-related children   that will not be visible to non-administrative users. | Bug | Major |   | ConfigAndInstallShouldOnlyContainOsgiNodes | aem
adobe/cq60:core.wcm.components.content:2.4.0 | 0 | Node   /apps/core/wcm/components/text/v1/text/cq:editConfig/cq:inplaceEditing/config   is an OSGi config or install path which contains non-OSGi-related children   that will not be visible to non-administrative users. | Bug | Major |   | ConfigAndInstallShouldOnlyContainOsgiNodes | aem


to fix this, you should add the [oakpal](http://adamcin.net/oakpal/oakpal-maven-plugin/usage.html) plugin and skip that particular check from the scan as follows:

1. Follow the [setup instructions](http://adamcin.net/oakpal/oakpal-maven-plugin/usage.html) to get the cnd. I chose to store it under `src/test/resources/node-type.cnd`

2. Add the plugin *in my case, to ui.apps pom file*

> note the `subpackages` check is skipped below.
```
 <plugin>
                <groupId>net.adamcin.oakpal</groupId>
                <artifactId>oakpal-maven-plugin</artifactId>
                <version>1.3.0</version>
                <configuration>
                    <cndNames>node-type.cnd</cndNames>
                    <checklists> <checklist>basic</checklist></checklists>
                    <checks>
                        <check>
                            <name>subpackages</name>
                            <skip>true</skip>
                        </check>
                    </checks>
                </configuration>
                <executions>
                    <execution>
                        <goals>
                            <goal>scan</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
```

and voila, cloud manager no longer complains.

**Comment by richardhand** [comment]
Reopening here as the issue still requires verification

**Comment by richardhand** [comment]
Code is merged to [master](https://github.com/adobe/aem-core-wcm-components/commits/master) and will be available in the upcoming [2.6.0 Release](https://github.com/adobe/aem-core-wcm-components/milestone/16).
